src/modules/Credential/containers/CredentialRecipientModal/useCredentialRecipientModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { addCredentialRecipientAdaptor, Schema, Attribute } from 'src/core/adaptors';
+import { CredentialClaims } from 'src/core/api';
+import { emailPattern, urlPattern } from 'src/core/helpers/regexs';
+import * as yup from 'yup';
+
+const recipientSchema = yup
+  .object()
+  .shape({
+    recipient_email: yup
+      .string()
+      .email()
+      .matches(emailPattern, 'This field must be a valid Email')
+      .required('This field is required'),
+    recipient_firstName: yup.string().required('This field is required'),
+    recipient_lastName: yup.string().required('This field is required'),
+  })
+  .required();
+
+const createValidationSchema = (schemaAttributes: Attribute[]) => {
+  const shape: { [key: string]: any } = {};
+  const validations = {
+    TEXT: yup.string().required('This field is required'),
+    NUMBER: yup.number().typeError('This field must be a number').required('This field is required'),
+    BOOLEAN: yup.boolean().default(false),
+    URL: yup.string().matches(urlPattern, 'This field must be a valid URL').required('This field is required'),
+    EMAIL: yup
+      .string()
+      .email()
+      .matches(emailPattern, 'This field must be a valid Email')
+      .required('This field is required'),
+    DATETIME: yup.date().typeError('This field must be a valid date').required('This field is required'),
+  };
+
+  schemaAttributes.forEach(attribute => {
+    shape[attribute.name] = validations[attribute.option.value];
+  });
+  const dynamicSchema = yup.object().shape(shape);
+  return dynamicSchema.concat(recipientSchema);
+};
+
+export const useCredentialRecipientModal = (selectedSchema: Schema, onAddCredentialRecipient: () => void) => {
+  const { t: translate } = useTranslation();
+  const schemaAttributes = selectedSchema?.attributes || [];
+  const validationSchema = createValidationSchema(schemaAttributes);
+  const {
+    control,
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    reset,
+  } = useForm({
+    mode: 'all',
+    resolver: yupResolver(validationSchema),
+  });
+  const disabledSubmit = !isValid;
+
+  const areEqual = (str1: string, str2: string) => {
+    return (
+      str1.replaceAll('_', '').replaceAll(' ', '').toUpperCase() ===
+      str2.replaceAll('_', '').replaceAll(' ', '').toUpperCase()
+    );
+  };
```

Comment:
We are using this function in different components, it may be better to move it to string utils

Reply:
Please see the deletion 
We changed the credentials issuance flow so this code is same as before. It just moved and replace into another component 
Not new one